packages/react-shopify-app-route-propagator/README.md

```diff
@@ -65,74 +30,135 @@ export interface Props {
 }
 ```
 
-Allowing you to use it with a string
+### `useRoutePropagator`
 
-```javascript
-<ShopifyRoutePropagator location="/foo/bar?thing=true" app={app} />
-```
-
-Or a location object
+This example uses [app bridge](https://help.shopify.com/en/api/embedded-apps/app-bridge#set-up-your-app) to create an app instance and `withRouter` HoC form `react-router-router`.
 
 ```javascript
-<ShopifyRoutePropagator location={this.props.location} app={app} />
-```
+import React from 'react';
+import {Switch, Route, withRouter} from 'react-router'
+import {BrowserRouter} from 'react-router-dom'
+import createApp, {getShopOrigin} from '@shopify/app-bridge';
+import {useRoutePropagator} from '@shopify/react-shopify-app-route-propagator';
 
-Or even with `window.location` (though then you will need to make sure it rerenders yourself)
+export default withRouter(function(props) {
+  useRoutePropagator(
+    createApp({
+      apiKey: 'API key from Shopify Partner Dashboard',
+      shopOrigin: getShopOrigin(),
+    }),
+    props.location
+  );
 
-```javascript
-<ShopifyRoutePropagator location={window.location} app={app} />
+  return (
+    <React.Fragment>
+      <Switch>
+        <Route exact path="/">
+        { /* other routes */ }
+      </Switch>
+    </React.Fragment>
+  );
+})
 ```
 
-### With React Router (withRouter)
+### `<RoutePropagator />`
 
-An easy way to mount the component with minimal configuration in a [react-router](https://github.com/ReactTraining/react-router) application is to first import the `withRouter` Higher-Order-Component and create a wrapped version of the propagator.
+This example assume that you have [polaris-react](https://github.com/Shopify/polaris-react) v3 or higher.
+Plus `withRouter` HoC form `react-router-router`.
```

Comment:
from

Reply:
react-router-dom

Reply:
I correct the typo but `withRouter` is actually from `react-router` thou?